### PR TITLE
Implement COMMON blocks - AST to ASR

### PR DESCRIPTION
**Pull request by Pranavchiku**
This PR translates the AST and generates ASR for common blocks. Now we have to create an ASR pass that will identify these the variables present in the common modules and replace them with the common block variables ( present in common module ).

### CODE REVIEW

**Comment by certik** [comment]
Something like this. Let's add tests and see how the ASR looks like.

File: tests/common3.f [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+            integer i(5)
+            common /sample/ A, B, i
+            print *, A, B
+        end subroutine
+
+        program common1
+            real A, B
+            integer c, d, e, f, g, h, i(5), j
+            common /sample/ A, B, i
+            common /c/ c /b/ d, e, f /c/ g, h, j
```
Comment (certik) [inline tests/common3.f:12]:
```suggestion
            common /c/ c, /b/ d, e, f, /c/ g, h, j
```
And put this into integration_tests, as regular free form code. Test with gfortran and lfortran.

**Comment by certik** [comment]
In the symbol table visitor, I would replace "real A" with "use common_block_sample, only: A", which in ASR is done using the ExternalSymbol. Then in the body visitor everything should just work, as it will lookup the symbol A, and will find the ExternalSymbol and use it appropriately.

**Comment by Pranavchiku** [comment]
I will be creating multiple PRs to implement this, feel free to populate this TODO list
* [ ] Add a new member in Variable Symbol `symbol? common_block`
* [ ] Represent the variables as a struct and perform AST to ASR implementation

**Comment by Pranavchiku** [comment]
Closing with reference to #1665.

**Pull request closed without merging**